language_service/src/compilation.rs

```diff
@@ -100,12 +85,19 @@ impl Compilation {
             unit,
         );
 
-        let lints = qsc::linter::run_lints(unit, Some(lints_config));
-        for lint in lints {
-            errors.push(WithSource::from_map(
-                &unit.sources,
-                qsc::compile::ErrorKind::Lint(lint),
-            ));
+        // Compute new lints and append them to the errors Vec.
```

Comment:
The changes in this file were to ensure RIR passes happen before lints. Since lints can add warnings and errors, and RIR will only run when errors list is empty, a lint warning would prevent RIR checks, which is not desireable.

Reply:
Looks like this is already getting fixed in #1439, so I can remove this part of the changes.